bndtools.m2e/src/bndtools/m2e/MavenImplicitProjectRepository.java

```diff
@@ -172,6 +173,20 @@ private void createRepo(IMavenProjectFacade projectFacade, IProgressMonitor moni
             if (logger.isErrorEnabled()) {
                 logger.error("Failed to create implicit repository for m2e project {}", getName(), e);
             }
+
+            try {
+                String name = mavenProject.getName()
+                    .isEmpty() ? mavenProject.getArtifactId() : mavenProject.getName();
+
+                fileSetRepository = new FileSetRepository(name, Collections.emptyList());
+
+                Central.refreshPlugin(this);
+            } catch (Exception e2) {
+                // we're really in trouble if we get here...
```

Comment:
Yes, but we still just eat the exception which means some NPE later. Why not just throw the (ducked) exception here? Then at least we have a more useful stack trace?